app/submitter/convert_payload_0_0_3.py

```diff
@@ -8,16 +14,56 @@ def convert_answers_to_payload_0_0_3(answer_store, schema, routing_path):
         },
         {
             'value': 'Husband or wife',
+            'list_item_id': 'abc123',
             'answer_id': 'who-is-related',
         }
     ]
-    :param answer_store: questionnaire answers
-    :param routing_path: the path followed in the questionnaire
-    :return: data in a formatted form
+
+    For list answers, this method will query the list store and get all answers from the
+    add list item block. If there are multiple list collectors for one list, they will have
+    the same answer_ids, and will not be duplicated.
+
+    Returns:
+        A list of answer dictionaries.
     """
-    data = []
+    temp_answers = AnswerStore()
```

Comment:
what is temp about these answers?

Reply:
I'll change the name.

I'm also not entirely happy with this solution. I wanted to use a set here, which removes a load of code from this loop. This doesn't work because we sometimes store lists as values in answers, which means the Answer object is not hashable. I suppose one solution here is to override __hash__ and do something hacky, but the list would still be mutable, so it's not ideal.